cmd/buildctl/build.go

```diff
@@ -291,6 +297,20 @@ func buildAction(clicontext *cli.Context) error {
 		// not using shared context to not disrupt display but let is finish reporting errors
 		return progressui.DisplaySolveStatus(context.TODO(), "", c, os.Stderr, displayCh)
 	})
+	err = eg.Wait()
+	return err
+}
 
-	return eg.Wait()
+func deleteExportedFiles(exports []client.ExportEntry) {
+	for _, ex := range exports {
+		switch ex.Type {
+		case client.ExporterOCI, client.ExporterDocker, client.ExporterTar:
+			dest := ex.Attrs["dest"]
+			fi, err := os.Stat(dest)
+			if err == nil && !fi.IsDir() {
+				logrus.Debugf("build got error, deleting temp file: %s", dest)
+				system.EnsureRemoveAll(dest)
```

Comment:
Vendor check is likely failing, because this adds `github.com/docker/docker/pkg/system` as a new dependency. Looking at this, I think this is just to remove a file, so I don't think this needs that package, and `os.Remove(dest)` would work? (but may require some additional error-handling if remove fails?)

Reply:
Your detailed comment is really helpful, thanks @thaJeztah !

Actually this `github.com/docker/docker/pkg/system` is already in folder vendor... But you reminds me that I should check `go.mod` and `go.sum` at the same time.

And I like `system.EnsureRemoveAll()` more than `os.Remove()`, it works really good in docker! But if with `os.Remove()`, I think we can just print a warning if remove fails?

Reply:
No... We can't use warning print, it will be printed on the progress UI 😂 
A debug print with err will be enough?

Reply:
`system.EnsureRemoveAll()` feels like a bit of overkill if it's a single file (and if we don't have to deal with the race-condition described on that function), but I haven't dug deep into how it's used in this repository.

Just a `logrus.Debugf()` or `logrus.Infof()` is probably ok (assuming we want to return the original error caused by the build itself)